packages/account/src/providers/transaction-summary/input.ts

```diff
@@ -33,16 +34,23 @@ export function getInputsContract(inputs: Input[]) {
 }
 
 /** @hidden */
-export function getInputFromAssetId(inputs: Input[], assetId: string) {
+export function getRelevantInputs(inputs: Input[], assetId: string = ZeroBytes32) {
   const coinInputs = getInputsCoin(inputs);
   const messageInputs = getInputsMessage(inputs);
-  const coinInput = coinInputs.find((i) => i.assetId === assetId);
-  // TODO: should include assetId in InputMessage as well. for now we're mocking ETH
-  const messageInput = messageInputs.find(
-    (_) => assetId === '0x0000000000000000000000000000000000000000000000000000000000000000'
-  );
+  let coinInput = coinInputs.find((i) => i.assetId === assetId);
+  // #TODO: There are times when the ReceiptCall's baseAssetId doesn't match the CoinInput's assetId
+  // In this case, we should return the last input in the CoinInput array
+  if (!coinInput && coinInputs.length > 0) {
+    coinInput = coinInputs[coinInputs.length - 1];
+  }
```

Comment:
I don't think this exception should be handled here. this function should purely return the correct input for the param asset id.

If the contract call receipt is the problem, the exception should be handled there before executing this function `getInputFromAssetId`